text/src/autogluon/text/text_prediction/mx/models.py

```diff
@@ -982,15 +1043,27 @@ def train(self, train_data, tuning_data,
             time_attr='report_idx',
             reward_attr='reward_attr',
             dist_ip_addrs=scheduler_options.get('dist_ip_addrs'))
+
         # Create a temporary cache file. The internal train function will load the
         # temporary cache.
-        os.makedirs(os.path.join(self._output_directory, 'data_cache'), exist_ok=True)
-        train_df_path = os.path.join(self._output_directory, 'data_cache',
-                                     'cache_train_dataframe.pd.pkl')
-        tuning_df_path = os.path.join(self._output_directory,  'data_cache',
-                                      'cache_tuning_dataframe.pd.pkl')
+        # In fact, we may generalize this functionality to create the cache in S3/FSx.
+        cache_path = os.path.join(self._output_directory, 'cache')
+        os.makedirs(cache_path, exist_ok=True)
+        train_df_path = os.path.join(cache_path, 'cache_train_dataframe.pd.pkl')
+        tuning_df_path = os.path.join(cache_path, 'cache_tuning_dataframe.pd.pkl')
         train_data.to_pickle(train_df_path)
         tuning_data.to_pickle(tuning_df_path)
+        if continue_training:
+            # We need to store the current weights to the local disk as temporary cache.
```

Comment:
wouldn't the weights already be stored on disk in this case? I.e. wasn't the model loaded from disk before training is continued?

Reply:
It's possible that the predictor is called twice and the model hasn't been saved.